frontend/model/contracts/group.js

```diff
@@ -1121,6 +1128,28 @@ sbp('chelonia/defineContract', {
       }
       await sbp('gi.db/archive/save', key, proposals)
       sbp('okTurtles.events/emit', PROPOSAL_ARCHIVED, [proposalHash, proposal])
+    },
+    'gi.contracts/group/archivePayments': async function (contractID, paymentsByPeriod) {
+      const { username } = sbp('state/vuex/state').loggedIn
+      for (const period of Object.keys(paymentsByPeriod).sort()) {
+        const periodKey = `paymentPeriods/${username}/${contractID}`
+        const periods = await sbp('gi.db/archive/load', periodKey) || []
+        const paymentsKey = `paymentsByPeriod/${username}/${contractID}/${period}`
+        const payments = await sbp('gi.db/archive/load', paymentsKey) || {}
+
+        periods.unshift(period)
+        merge(payments, paymentsByPeriod[period])
+
+        if (periods.length > MAX_ARCHIVED_PAYMENTS) {
```

Comment:
> Something like MAX_ARCHIVED_PERIODS.
Correct. That's more precise meaning.

> Also, since this value can change, this should be a while instead of an if.
I understand. Could be changed later like 20 to 10.